src/Common/dataAccess/readCollection.ts

```diff
@@ -22,3 +26,51 @@ export async function readCollection(databaseId: string, collectionId: string):
   clearMessage();
   return collection;
 }
+
+export async function getMongoCollectionIndexTransformationProgress(
+  databaseId: string,
+  collectionId: string
+): Promise<number> {
+  let indexTransformationPercentage: number;
+  const clearMessage = logConsoleProgress(`Reading container ${collectionId}`);
+  try {
+    const response = await client()
+      .database(databaseId)
+      .container(collectionId)
+      .read({ populateQuotaInfo: true });
+
+    indexTransformationPercentage = parseInt(
+      response.headers[Constants.HttpHeaders.collectionIndexTransformationProgress] as string
+    );
+  } catch (error) {
+    logConsoleError(`Error while reading container ${collectionId}:\n ${JSON.stringify(error)}`);
```

Comment:
>logConsoleError [](start = 4, length = 15)

what is the philosophy behind console errors?

Reply:
this I am not sure of. I reused existing patterns.

Reply:
victor added some error handling. I pulled those changes